### DISCUSSION TITLE
Incorporating CGAL into an existing QT6 project

### DISCUSSION THREAD

**Discussion by DarkSun812** [CGAL/cgal#8565] [Q&A] [answered]
I have a QT6 project with a .pro file and I want to include the CGAL libraries I saw in a StackOverflow question where a person asked the same question but in linux and I can't find the libs folder in the windows installation folder.

The link to the question I found from stack is this: https://stackoverflow.com/questions/8524217/how-do-you-include-cgal-into-a-qt-project-in-qt-creator

This is my .pro file but as you can imagine, I don't have the usr/include folder so I don't know how to proceed.

```qmake
# TestCGAL.pro
CONFIG += c++17

# You can make your code fail to compile if it uses deprecated APIs.
# In order to do so, uncomment the following line.
#DEFINES += QT_DISABLE_DEPRECATED_BEFORE=0x060000    # disables all the APIs deprecated before Qt 6.0.0

SOURCES += \
    main.cpp \
    mainwindow.cpp

HEADERS += \
    mainwindow.h

INCLUDEPATH +=   C:/dev/CGAL-6.0/include\
LIBS        += -L/usr/include/
LIBS        += -lCGAL
LIBS        += -lgmp
LIBS        += -lmpfr // not really needed for me, but added since gmp had to be added too
QMAKE_CXXFLAGS += -frounding-math -O3

FORMS += \
    mainwindow.ui

# Default rules for deployment.
qnx: target.path = /tmp/$${TARGET}/bin
else: unix:!android: target.path = /opt/$${TARGET}/bin
!isEmpty(target.path): INSTALLS += target

```

**Comment by lrineau** [comment]
As far as I know, there is not any CGAL developer that uses `qmake` and `.pro` files. The CGAL project supports only the use of CMake files.

You will probably not get any useful answer here. I am sorry.

**Reply by DarkSun812** [reply]
If that is the case, where can I get information on how to incorporate CGAl into a project using QMake?

**Reply by lrineau** [reply]
Try have a look at qmake documentation: https://doc.qt.io/qt-6/qmake-manual.html. You might find the answers you need.

Your `LIBS` settings are wrong, for a Windows project: https://doc.qt.io/qt-6/qmake-variable-reference.html#libs

And maybe remove the last `\` character from `INCLUDEPATH`.

**Reply by DarkSun812** [reply]
Thanks, if I want to switch from QMake to CMake, how can I do it? Because I use windows and as far as I understand CMake is for linux, so I have had many problems to use it from the beginning, so I used QMake at first, but I am open to the option of moving to CMake because I understand that it is faster than QMake.

**Reply by lrineau** [reply, marked as answer]
CMake is cross-platform. That is the reason why it was adopted by CGAL.

I suggest you have a lot at the documentation. A few references:

  - [Getting Started with CGAL](https://doc.cgal.org/latest/Manual/general_intro.html),
  - [CMake documentation](https://cmake.org/cmake/help/latest/),
  - [CMake Tutorial](https://cmake.org/cmake/help/latest/guide/tutorial/index.html).
